crypto/src/main/scala/org/bitcoins/crypto/MuSig2Util.scala

```diff
@@ -0,0 +1,225 @@
+package org.bitcoins.crypto
+
+import scodec.bits.ByteVector
+
+// TODO partial signature verification
+// TODO implement tweaking
+object MuSig2Util {
+
+  private val nonceNum: Int = 2
+
+  type MultiNoncePriv = Vector[ECPrivateKey]
+  type MultiNoncePub = Vector[ECPublicKey]
+
+  case class KeyGenContext(
+      aggPubKey: ECPublicKey,
+      tacc: FieldElement,
+      gacc: Boolean) {
+    def getPubKey: ByteVector = aggPubKey.bytes.tail
+    def getPubKey33: ByteVector = aggPubKey.bytes
+  }
+
+  case class KeySet(keys: Vector[SchnorrPublicKey]) {
+    keys.init.zip(keys.tail).foreach { case (key1, key2) =>
+      require(key1.hex.compareTo(key2.hex) <= 0,
+              "Keys must be sorted lexicographically")
+    }
```

Comment:
might be eaiser to make a `Ordering[SchnorrPublicKey]` so you can just call `.sorted`

Reply:
Perhaps we should just have a lexicographic ordering defined for all NetworkElements. Now that I say this I vaguely recall already doing this but maybe it's false deja vu, I'll look for it